tests/parser/ast_utils/test_ast_dict.py

```diff
@@ -0,0 +1,81 @@
+from vyper import (
+    compiler,
+)
+from vyper.ast_utils import (
+    ast_to_dict,
+    dict_to_ast,
+    parse_to_ast,
+)
+
+
+def get_node_ids(ast_struct, ids=None):
+    if ids is None:
+        ids = []
+
+    for k, v in ast_struct.items():
+        print(k, v, ids)
+        if isinstance(v, dict):
+            ids = get_node_ids(v, ids)
+        elif isinstance(v, list):
+            for x in v:
+                ids = get_node_ids(x, ids)
+        elif k == 'node_id':
+            ids.append(v)
```

Comment:
This will silently continue if none of these conditions are met.  Consider a final `else` clause that raises an exception.